### PR TITLE
Add via support for the Leopold FC980C

### PR DESCRIPTION
**Pull request by penmoid**
## Description

This pull request adds Via support to the FC980C. The board's config.h file was updated to match the manufacturer and model naming convention from the FC660C, which already has Via support. A Via keymap was added, and due to EEPROM size constraints the configuration has been limited to 3 dynamic layers. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/fc980c/keymaps/via/README.md
Comment (drashna) [inline keyboards/fc980c/keymaps/via/README.md:1]: Please remove this file.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/fc980c/keymaps/via/README.md
Comment (penmoid) [reply keyboards/fc980c/keymaps/via/README.md]: Removed file in latest commit

**Review by drashna — APPROVED** [review]

**Comment by penmoid** [comment]
@drashna Does this require further review before it can be merged?

**Comment by drashna** [comment]
Yeah. Generally, we want two reviews before merging.  Mostly, to make sure that no one set of eyes has missed anything.  It still happens, but two sets makes it much less likely.

File: keyboards/fc980c/keymaps/via/config.h
Comment (fauxpark) [inline keyboards/fc980c/keymaps/via/config.h:22]: Should probably go in the keyboard-level config.h so that it affects everyone who enables VIA on their own keymaps.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/fc980c/keymaps/via/config.h
Comment (penmoid) [reply keyboards/fc980c/keymaps/via/config.h]: This change has been made. Validated that the firmware still compiles and functions properly.

File: keyboards/fc980c/keymaps/via/config.h
Comment (fauxpark) [inline keyboards/fc980c/keymaps/via/config.h:1]: This file can be deleted now.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/fc980c/keymaps/via/config.h
Comment (penmoid) [reply keyboards/fc980c/keymaps/via/config.h]: Removed in latest commit.

**Comment by penmoid** [comment]
@fauxpark Are there any remaining changes that need to be made in this PR?

**Review by fauxpark — APPROVED** [review]

**Comment by penmoid** [comment]
@fauxpark @drashna Would it be possible to have this merged now that is has been approved? It is waiting on approval to run a workflow.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**
